pkg/controller/gitrepo/gitrepo_reconcile.go

```diff
@@ -66,11 +66,12 @@ func (r *ReconcileGitRepo) Reconcile(request reconcile.Request) (reconcile.Resul
 		},
 		DeployKeys: instance.Spec.DeployKeys,
 		Logger:     reqLogger,
+		BaseURL:    string(secret.Data[SecretEndpointName]),
```

Comment:
Let's use a `URL` instead of string here. Also check if the secrets contains a `endpoint` and `token`.